bedita-app/vendors/shells/dbadmin.php

```diff
@@ -859,31 +859,78 @@ public function clearMediaCache() {
 	public function massRemove() {
 
 		if(empty($this->params['type'])) {
-			$this->out("Parameter -type mandatory [object type]");
+			$this->out('Parameter -type mandatory [object type]');
 			return;
 		}
-
+		// force delete (it continues, on errors)
+		$force = (!empty($this->params['f']));
+		// max number of elements to delete at each iterations
+		$limit = (!empty($this->params['n'])) ? $this->params['n'] : 100;
 		$type = $this->params['type'];
-		$objTypeId = Configure::read("objectTypes." . $type . ".id");
+		$objTypeId = Configure::read('objectTypes.' . $type . '.id');
 		if(empty($objTypeId)) {
-			$this->out("object type " . $type . " not found");
+			$this->out('object type ' . $type . ' not found');
 		}
 
-		$model = ClassRegistry::init("BEObject");
+		$model = ClassRegistry::init('BEObject');
 
-		$modelType = Configure::read("objectTypes." . $type . ".model");
-		$this->out("Removing all " . $modelType ." from your instance");
-		$ans = $this->in("Proceed??? [y/n]");
-		if($ans != "y") {
-			$this->out("Bye");
+		$modelType = Configure::read('objectTypes.' . $type . '.model');
+		$this->out('Removing all ' . $modelType .' from your instance');
+		$ans = $this->in('Proceed??? [y/n]');
+		if($ans != 'y') {
+			$this->out('Bye');
 			return;
 		}
-		$res = $model->deleteAll("BEObject.object_type_id = '$objTypeId'");
-		if($res == false) {
-			$this->out("Error removing items");
-			return;
+		$counter = 0;
+		$count = $model->find('count', array('conditions' => array('BEObject.object_type_id' => $objTypeId)));
+		$skipids = array();
+		$conditions = array('BEObject.object_type_id' => $objTypeId);
+		while ($count > 0) {
```

Comment:
It should be better use a transaction to avoid data inconsistency if some delete fails.